### PR TITLE
Fix cli tests

### LINKED ISSUE #7078: CI failing on master: External error: query is expected to fail, but actually succeed:
### Describe the bug

https://github.com/apache/arrow-datafusion/actions/runs/5647552111/job/15298151230

```
External error: query is expected to fail, but actually succeed:
[SQL] SELECT i - ts1 from FOO;
at tests/sqllogictests/test_files/timestamps.slt:1253

External error: query is expected to fail, but actually succeed:
[SQL] SELECT array_agg(c13 ORDER BY c1) FROM aggregate_test_100
at tests/sqllogictests/test_files/aggregate.slt:99

External error: query is expected to fail, but actually succeed:
[SQL] SELECT s.country, ARRAY_AGG(s.amount ORDER BY s.amount DESC, s.country DESC) AS amounts,
    SUM(s.amount ORDER BY s.amount DESC) AS sum1
  FROM sales_global AS s
  GROUP BY s.country
at tests/sqllogictests/test_files/groupby.slt:2229

External error: query is expected to fail, but actually succeed:
[SQL] SELECT
SUM(c4) OVER(ORDER BY c2 RANGE BETWEEN 1 PRECEDING AND 1 FOLLOWING),
SUM(c3) OVER(ORDER BY c2 RANGE BETWEEN 10000 PRECEDING AND 10000 FOLLOWING),
COUNT(*) OVER(ORDER BY c2 RANGE BETWEEN 1 PRECEDING AND 1 FOLLOWING)
FROM aggregate_test_100
ORDER BY c9
LIMIT 5
at tests/sqllogictests/test_files/window.slt:718
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mustafasrepo**
# Which issue does this PR close?

Closes #7078. 

# Rationale for this change
With the new changes in sqllogictest, failing tests are asserted also.(Previously, if some tests were supposed fail but succeeds there was no error.) See [discussion](https://github.com/apache/arrow-datafusion/issues/7078#issuecomment-1649209664). Thanks @alamb and @jonahgao for finding out root cause of the error.

This PR fixes this problem. 
All of the tests(except the test in `timestamp.slt`) supposed to run correctly. Expected errors are leftovers from the time when these features are not supported. However, as far as I know, the test in `timestamp.slt` should fail. I have commented that test out, and marked it as TODO. Fixing it outside the scope of this PR.

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 1]
```diff
@@ -95,9 +95,16 @@ SELECT approx_percentile_cont(c3, 0.95, c1) FROM aggregate_test_100
 statement error Error during planning: The percentile sample points count for ApproxPercentileCont must be integer, not Float64.
 SELECT approx_percentile_cont(c3, 0.95, 111.1) FROM aggregate_test_100
 
-# csv_query_array_agg_unsupported
-statement error This feature is not implemented: Order-sensitive aggregators is not supported on multiple partitions
-SELECT array_agg(c13 ORDER BY c1) FROM aggregate_test_100
+# array agg can use order by
```
Comment (mustafasrepo) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:98]: To make result deterministic, I have changed the test

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 2]
```diff
@@ -1223,10 +1230,6 @@ select c2, sum(c3) sum_c3, avg(c3) avg_c3, max(c3) max_c3, min(c3) min_c3, count
 4 29 1.260869565217 123 -117 23
 5 -194 -13.857142857143 118 -101 14
 
-# csv_query_array_agg_unsupported
-statement error This feature is not implemented: Order-sensitive aggregators is not supported on multiple partitions
-SELECT array_agg(c13 ORDER BY c1) FROM aggregate_test_100;
-
```
Comment (mustafasrepo) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1229]: This is the same test with above, hence I deleted it

File: datafusion/core/tests/sqllogictests/test_files/groupby.slt [hunk 3]
```diff
@@ -2222,16 +2222,6 @@ FRA [200.0, 50.0] 250
...
 
-# test_ordering_sensitive_multiple_req
-# Currently we do not support multiple ordering requirement for aggregation
-# once this support is added. This test should change
-# See issue: https://github.com/sqlparser-rs/sqlparser-rs/issues/875
-statement error DataFusion error: This feature is not implemented: ARRAY_AGG only supports a single ORDER BY expression\. Got 2
-SELECT s.country, ARRAY_AGG(s.amount ORDER BY s.amount DESC, s.country DESC) AS amounts,
-    SUM(s.amount ORDER BY s.amount DESC) AS sum1
-  FROM sales_global AS s
-  GROUP BY s.country
```
Comment (mustafasrepo) [inline datafusion/core/tests/sqllogictests/test_files/groupby.slt:2233]: There are already similar test existing tests for this functionality. To not duplicate same/similar tests. I have removed this test also.

File: datafusion/core/tests/sqllogictests/test_files/window.slt [hunk 4]
```diff
@@ -714,15 +714,27 @@ LIMIT 5
...
 
 
-#fn window_frame_ranges_preceding_following
-statement error DataFusion error: Internal error: Operator \- is not implemented for types
+# fn window_frame_ranges_preceding_following
+# when value is outside type range (10000 is outside range of tiny int (type of c2)),
+# we should treat values as infinite, hence
+# "SUM(c3) OVER(ORDER BY c2 RANGE BETWEEN 10000 PRECEDING AND 10000 FOLLOWING),"
+# is functionally equivalent to
+# "SUM(c3) OVER(ORDER BY c2 RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING),"
```
Comment (mustafasrepo) [inline datafusion/core/tests/sqllogictests/test_files/window.slt:722]: This test supposed to run correctly, I have updated comment to explain how this query supposed to work.

File: datafusion/core/tests/sqllogictests/test_files/window.slt [hunk 5]
```diff
@@ -928,23 +940,6 @@ FROM aggregate_test_100
 ORDER BY c9
 LIMIT 5
 
-
-#fn window_frame_groups_preceding_following_desc
```
Comment (mustafasrepo) [inline datafusion/core/tests/sqllogictests/test_files/window.slt:932]: This test, is same with test above. We can remove this test

**Review by alamb — APPROVED** [review]
Thank you @mustafasrepo

**Comment by alamb** [comment]
I plan to merge this PR once the CI has passed

File: datafusion/core/tests/sqllogictests/test_files/timestamps.slt [hunk 6]
```diff
@@ -1250,8 +1250,10 @@ SELECT '2000-01-01T00:00:00'::timestamp - '2010-01-01T00:00:00'::timestamp;
 0 years 0 mons -3653 days 0 hours 0 mins 0.000000000 secs
 
 # Interval - Timestamp => error
-statement error DataFusion error: type_coercion\ncaused by\nError during planning: Interval\(MonthDayNano\) \- Timestamp\(Nanosecond, None\) can't be evaluated because there isn't a common type to coerce the types to
-SELECT i - ts1 from FOO;
+# statement error DataFusion error: type_coercion\ncaused by\nError during planning: Interval\(MonthDayNano\) \- Timestamp\(Nanosecond, None\) can't be evaluated because there isn't a common type to coerce the types to
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/timestamps.slt:1253]: filed https://github.com/apache/arrow-datafusion/pull/7085 to add a ticket reference here

**Review by jackwener — APPROVED** [review]
Thanks @mustafasrepo

**Pull request merged**